<launch>
    <arg name="namespace" default="rsgm_ros"/>
    <param name="use_sim_time" value="true"/>
    
    <node pkg="tf" type="static_transform_publisher" name="map2odom" args="0 10 0 0 0 0 map base_left_cam 100" />
    
    <group ns="$(arg namespace)">
        <node name="stereo_and_odom" pkg="polar_grid_tracking" type="stereo_and_odom" output="screen" required="true" >
        </node> 
        
        <node pkg="rsgm_ros" type="rsgm_ros" name="rsgm_ros" output="screen" required="true">
            <remap from="stereo" to="stereo_and_odom" />
            <remap from="stereo_and_odom/left/$(arg namespace)/image" to="stereo_and_odom/left/image" />
            <remap from="stereo_and_odom/right/$(arg namespace)/image" to="stereo_and_odom/right/image" />
            
            <param name="approximate_sync" value="true" />
            
<!--             sgm_5x5, hwcs_census_9x7 -->
            <param name="disparity_method" value="hwcs_census_9x7" />
<!--             standard_sgm, striped_sgm, striped_sgm_subsample_2, striped_sgm_subsample_4 -->
            <param name="sampling_method" value="striped_sgm" />
            
            <param name="paths" value="8" />
            <param name="threads" value="4" />
            <param name="strips" value="4" />
            <param name="disp_count" value="64" />
            
            <param name="downsample" value="false" />
        </node>
    </group>
    
    <node pkg="rviz" type="rviz" name="rviz" required="false"  args="&#45;&#45;display-config 
          $(find rsgm_ros)/config/rsgm_ros.rviz"/>
</launch>